src/main/java/picard/sam/SamToFastqWithTags.java

```diff
@@ -0,0 +1,237 @@
+    static final String USAGE_DETAILS = "Extracts read sequences and qualities from the input SAM/BAM file and SAM/BAM tags" +
+            " and writes them into the output file in Sanger FASTQ format." +
+            "See <a href=\"http://maq.sourceforge.net/fastq.shtml\">MAQ FASTQ specification</a> for details.<br /> <br />" +
+            "The following example will create two FASTQs from tags.  One will be converted with the base sequence coming from " +
+            "the \"CR\" tag and base quality from the \"CY\" tag.  The other fastq will be converted with the base sequence coming" +
+            " from the \"CB\" and \"UR\" tags concatenated together with no separator (not specified on command line) with the base" +
+            " qualities coming from the \"CY\" and \"UY\" tags concatenated together.  The two files will be named CR.fastq" +
+            " and CB_UR.fastq." +
+            "<br />" +
+            "<pre>" +
+            "java -jar picard.jar SamToFastqWithTags <br />" +
+            "     I=input.bam<br />" +
+            "     FASTQ=output.fastq<br />" +
+            "     SEQUENCE_TAG_GROUP=CR<br />" +
+            "     QUALITY_TAG_GROUP=CY<br />" +
+            "     SEQUENCE_TAG_GROUP=\"CB,UR\"<br />" +
+            "     QUALITY_TAG_GROUP=\"CY,UY\"" +
+            "</pre>" +
+            "<hr />";
+
+    @Argument(shortName = "STG", doc = "List of comma separated tag values to extract from Input SAM/BAM to be used as read sequence", minElements = 1)
+    public List<String> SEQUENCE_TAG_GROUP;
+
+    @Argument(shortName = "QTG", doc = "List of comma separated tag values to extract from Input SAM/BAM to be used as read qualities", optional = true)
+    public List<String> QUALITY_TAG_GROUP;
+
+    @Argument(shortName = "SEP", doc = "List of any sequences (e.g. 'AACCTG`) to put in between each comma separated list of sequence tags in each SEQUENCE_TAG_GROUP (STG)", optional = true)
+    public List<String> TAG_GROUP_SEPERATOR;
+
+    @Argument(shortName = "GZOPTG", doc = "Compress output FASTQ files per Tag grouping using gzip and append a .gz extension to the file names.")
+    public Boolean COMPRESS_OUTPUTS_PER_TAG_GROUP = false;
+
+    private final Log log = Log.getInstance(SamToFastqWithTags.class);
+
+    private static final String TAG_SPLIT_DEFAULT_SEP = "";
+    private static final String TAG_SPLIT_QUAL = "~";
+
+    private ArrayList<String[]> SPLIT_SEQUENCE_TAGS;
+    private ArrayList<String[]> SPLIT_QUALITY_TAGS;
+    private ArrayList<String> SPLIT_SEPARATOR_TAGS;
+
+    @Override
+    protected void initializeAdditionalWriters() {
+        setupTagSplitValues();
+    }
+
+    @Override
+    protected void handleAdditionalRecords(SAMRecord currentRecord, Map<SAMReadGroupRecord, List<FastqWriter>> tagWriters, SAMRecord read1, SAMRecord read2) {
+        if (currentRecord.isSecondaryOrSupplementary() && !INCLUDE_NON_PRIMARY_ALIGNMENTS) {
```

Comment:
These appear to duplicate checks done in `SamToFastq`. Are they needed here too?